File: src/FrameIO/Models/ComDeviceConfigModels.cs
using FrameIO.Data;
using System;
using static FrameHelper.EntityHelpers;

namespace FrameIO.Models
{
    public interface IComDeviceConfigDTO : IWithIdAndTimeStamp
    {
        string DeviceName { get; set; }
    }

    public class ComDeviceConfigDTO : IComDeviceConfigDTO
    {
        public Guid Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public string DeviceName { get; set; }
    }

    public interface IComDeviceConfigView : IComDeviceConfigDTO
    {
    }

    public class ComDeviceConfigView : ComDeviceConfigDTO, IComDeviceConfigView
    {
    }

}

File: src/FrameUI/src/app/models/ComDeviceConfigModels.ts

export interface IcomDeviceConfigDTO {
    id: string;
    timeStamp: Date;
    deviceName: string;
    
}

export class ComDeviceConfigDTO implements IcomDeviceConfigDTO {
    public id: string;
    
    public timeStamp: Date;
    
    public deviceName: string;
    
    constructor(model?: IcomDeviceConfigDTO) {
        if(model) {
            this.id = model.id;
            this.timeStamp = model.timeStamp;
            this.deviceName = model.deviceName;
            
        }
    }
}

export interface IcomDeviceConfigView extends IcomDeviceConfigDTO {
    
}

export class ComDeviceConfigView extends ComDeviceConfigDTO implements IcomDeviceConfigView {
    constructor(model?: IcomDeviceConfigView) {
	super(model);

        if(model) {
            
        }
    }
}
 